### PR TITLE
 Remove unnecessary casts from OrtValue to MLValue

### PR DESCRIPTION
**Pull request by snnn**
1. Remove unnecessary casts from OrtValue to MLValue
2. Rename MLValue to OrtValue (not finished. will do more in another PR)

### CODE REVIEW

File: onnxruntime/core/framework/session_state_initializer.cc [hunk 1]
```diff
@@ -224,7 +224,7 @@ static common::Status DeserializeTensorProto(const Env& env, const std::basic_st
   OrtAllocatorInfo info(CPU, OrtDeviceAllocator, 0, OrtMemTypeDefault);
   std::unique_ptr<char[]> data(new char[cpu_tensor_length]);
   std::unique_ptr<Tensor> p_tensor;
-  MLValue tmp_mlvalue;
+  OrtValue tmp_mlvalue;
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/session_state_initializer.cc]: 'tmp_mlvalue' should be renamed now

File: onnxruntime/core/framework/session_state_initializer.cc [hunk 2]
```diff
@@ -187,13 +187,13 @@ common::Status SaveMLValueNameIndexMapping(const GraphViewer& graph_viewer,
...
 
-  LOGS(logger, INFO) << "Done saving MLValue mappings.";
+  LOGS(logger, INFO) << "Done saving OrtValue mappings.";
   return Status::OK();
 }
 
 static common::Status DeserializeTensorProto(const Env& env, const std::basic_string<PATH_CHAR_TYPE>& proto_path,
                                              const ONNX_NAMESPACE::TensorProto& tensor_proto, const MemBuffer& m,
-                                             const ExecutionProviders& exec_providers, MLValue& mlvalue,
+                                             const ExecutionProviders& exec_providers, OrtValue& mlvalue,
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/session_state_initializer.cc]: Rename parameter

File: onnxruntime/core/framework/session_state.h [hunk 3]
```diff
@@ -71,17 +71,17 @@ class SessionState {
   // initialized tensors
   /**
   * Adds an initialized tensor (weight) so that it can be used by the
-  * execution frame to setup the appropriate MLValue vectors.
+  * execution frame to setup the appropriate OrtValue vectors.
   * This function will take a shallow copy of d if d is not NULL
   */
-  Status AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue, const OrtCallback* d);
+  Status AddInitializedTensor(int mlvalue_index, const OrtValue& mlvalue, const OrtCallback* d);
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/session_state.h]: Rename parameter

File: onnxruntime/core/framework/session_state.cc [hunk 4]
```diff
@@ -36,7 +36,7 @@ void SessionState::SetExecutionPlan(std::unique_ptr<SequentialExecutionPlan> p_s
 
 const SequentialExecutionPlan* SessionState::GetExecutionPlan() const { return p_seq_exec_plan_.get(); }
 
-Status SessionState::AddInitializedTensor(int mlvalue_index, const MLValue& mlvalue, const OrtCallback* d) {
+Status SessionState::AddInitializedTensor(int mlvalue_index, const OrtValue& mlvalue, const OrtCallback* d) {
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/session_state.cc]: Rename parameter

File: onnxruntime/core/framework/execution_frame.h [hunk 5]
```diff
@@ -116,14 +116,14 @@ class ExecutionFrame final : public IExecutionFrame {
   // TODO: These two AllocateMLValue... methods are in the API purely for unit test usage.
   // Fix the unit tests so they set an execution plan that results in these methods being called by
   // GetOrCreateNodeOutputMLValue instead
-  Status AllocateMLValueTensorSelfOwnBuffer(MLValue& mlvalue,
+  Status AllocateMLValueTensorSelfOwnBuffer(OrtValue& mlvalue,
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename parameter.
Rename function

File: onnxruntime/core/framework/execution_frame.h [hunk 6]
```diff
@@ -116,14 +116,14 @@ class ExecutionFrame final : public IExecutionFrame {
...
                                             int mlvalue_index,
                                             MLDataType element_type,
                                             const OrtAllocatorInfo& location,
                                             const TensorShape& shape,
                                             bool create_fence = false);
 
-  Status AllocateMLValueTensorPreAllocateBuffer(MLValue& mlvalue,
+  Status AllocateMLValueTensorPreAllocateBuffer(OrtValue& mlvalue,
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename parameter
Rename function

File: onnxruntime/core/framework/execution_frame.h [hunk 7]
```diff
@@ -78,35 +78,35 @@ class IExecutionFrame {
...
   // Input and Output values are passed in by executors
-  std::vector<MLValue> all_values_;
+  std::vector<OrtValue> all_values_;
 
   const std::vector<int> fetch_mlvalue_idxs_;
 };
 
 class ExecutionFrame final : public IExecutionFrame {
  public:
   ExecutionFrame(const std::vector<int>& feed_mlvalue_idxs,
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename 'feed_mlvalue_idxs'

File: onnxruntime/core/framework/execution_frame.h [hunk 8]
```diff
@@ -78,35 +78,35 @@ class IExecutionFrame {
...
 
-  const MLValue& GetMLValue(int mlvalue_index) const {
+  const OrtValue& GetMLValue(int mlvalue_index) const {
     ORT_ENFORCE(mlvalue_index >= 0 && static_cast<size_t>(mlvalue_index) < all_values_.size());
     return all_values_[mlvalue_index];
   }
 
   virtual AllocatorPtr GetAllocatorImpl(const OrtAllocatorInfo& info) const = 0;
-  virtual Status CreateNodeOutputMLValueImpl(MLValue& mlvalue, int mlvalue_idx, const TensorShape* shape) = 0;
+  virtual Status CreateNodeOutputMLValueImpl(OrtValue& mlvalue, int mlvalue_idx, const TensorShape* shape) = 0;
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename parameter + function name

File: onnxruntime/core/framework/execution_frame.h [hunk 9]
```diff
@@ -78,35 +78,35 @@ class IExecutionFrame {
...
-            const std::unordered_map<int, MLValue>& initializers,
+            const std::vector<OrtValue>& feeds,
+            const std::unordered_map<int, OrtValue>& initializers,
             const std::vector<int>& fetch_mlvalue_idxs,
-            const std::vector<MLValue>& fetches,
+            const std::vector<OrtValue>& fetches,
             const MLValueNameIdxMap& mlvalue_idx_map);
 
-  const MLValue& GetMLValue(int mlvalue_index) const {
+  const OrtValue& GetMLValue(int mlvalue_index) const {
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename function + parameter

File: onnxruntime/core/framework/execution_frame.h [hunk 10]
```diff
@@ -65,8 +65,8 @@ class IExecutionFrame {
   // get the mlvalue_idx from NodeIndexInfo
   int GetNodeIdxToMLValueIdx(int index) const;
 
-  MLValue& GetMutableMLValue(int mlvalue_index) {
-    return const_cast<MLValue&>(GetMLValue(mlvalue_index));
+  OrtValue& GetMutableMLValue(int mlvalue_index) {
+    return const_cast<OrtValue&>(GetMLValue(mlvalue_index));
   }
 
   virtual Status ReleaseMLValueImpl(int mlvalue_idx);
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename function + parameter

File: onnxruntime/core/framework/execution_frame.h [hunk 11]
```diff
@@ -26,10 +26,10 @@ class NodeIndexInfo;
 class IExecutionFrame {
  protected:
   IExecutionFrame(const std::vector<int>& feed_mlvalue_idxs,
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.h]: Rename 'feed_mlvalue_idxs'

File: onnxruntime/core/framework/execution_frame.cc [hunk 12]
```diff
@@ -425,7 +425,7 @@ AllocatorPtr ExecutionFrame::GetAllocatorImpl(const OrtAllocatorInfo& info) cons
 
 // This method is not thread safe!
 // Return S_OK and nullptr if index map to an value that is an unused optional input/output
-Status ExecutionFrame::CreateNodeOutputMLValueImpl(MLValue& mlvalue, int mlvalue_idx, const TensorShape* shape) {
+Status ExecutionFrame::CreateNodeOutputMLValueImpl(OrtValue& mlvalue, int mlvalue_idx, const TensorShape* shape) {
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.cc]: Rename parameter + function

File: onnxruntime/core/framework/execution_frame.cc [hunk 13]
```diff
@@ -351,14 +351,14 @@ Status ExecutionFrame::AllocateTensorWithPreAllocateBufferHelper(MLValue& mlvalu
...
-static Status AllocateTraditionalMLValue(MLValue& mlvalue, const NonTensorTypeBase& type) {
+static Status AllocateTraditionalMLValue(OrtValue& mlvalue, const NonTensorTypeBase& type) {
   auto creator = type.GetCreateFunc();
   mlvalue.Init(creator(), &type, type.GetDeleteFunc());
   return Status::OK();
 }
 
 // This method is not thread safe!
-Status ExecutionFrame::AllocateAsPerAllocationPlan(MLValue& mlvalue, int mlvalue_index, const TensorShape* shape) {
+Status ExecutionFrame::AllocateAsPerAllocationPlan(OrtValue& mlvalue, int mlvalue_index, const TensorShape* shape) {
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.cc]: Rename parameter

File: onnxruntime/core/framework/execution_frame.cc [hunk 14]
```diff
@@ -351,14 +351,14 @@ Status ExecutionFrame::AllocateTensorWithPreAllocateBufferHelper(MLValue& mlvalu
   return Status::OK();
 }
 
-static Status AllocateTraditionalMLValue(MLValue& mlvalue, const NonTensorTypeBase& type) {
+static Status AllocateTraditionalMLValue(OrtValue& mlvalue, const NonTensorTypeBase& type) {
```
Comment (RyanUnderhill) [inline onnxruntime/core/framework/execution_frame.cc]: Rename function + parameter

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 15]
```diff
@@ -264,11 +264,11 @@ ORT_API_STATUS_IMPL(OrtCreateTensorWithDataAsOrtValue, _In_ const OrtAllocatorIn
       return OrtCreateStatus(ORT_NOT_IMPLEMENTED, errmsg.c_str());
     }
   }
-  std::unique_ptr<MLValue> value = std::make_unique<MLValue>();
+  std::unique_ptr<OrtValue> value = std::make_unique<OrtValue>();
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 16]
```diff
@@ -330,11 +330,11 @@ ORT_API_STATUS_IMPL(OrtCreateTensorAsOrtValue, _Inout_ OrtAllocator* allocator,
...
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 17]
```diff
@@ -532,7 +532,7 @@ ORT_API_STATUS_IMPL(OrtTensorProtoToOrtValue, _In_ const void* input, int input_
   if (!proto.ParseFromArray(input, input_len)) {
     return OrtCreateStatus(ORT_FAIL, "parse input tensor proto failed");
   }
-  std::unique_ptr<MLValue> value = std::make_unique<MLValue>();
+  std::unique_ptr<OrtValue> value = std::make_unique<OrtValue>();
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 18]
```diff
@@ -750,11 +750,11 @@ static OrtStatus* OrtGetValueImplSeqOfMap(const MLValue* p_ml_value, int index,
   auto& data_vec = p_ml_value->Get<T>();
   auto& data_elem = data_vec.at(index);
   auto copy_data_elem = std::make_unique<MapType>(data_elem);
-  std::unique_ptr<MLValue> value = std::make_unique<MLValue>();
+  std::unique_ptr<OrtValue> value = std::make_unique<OrtValue>();
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 19]
```diff
@@ -936,11 +936,11 @@ static OrtStatus* OrtCreateValueImplSeqHelperMap(OrtValue** const in, size_t num
     vec_ptr->push_back(m);
   }
   // create MLValue with this vector
-  std::unique_ptr<MLValue> value = std::make_unique<MLValue>();
+  std::unique_ptr<OrtValue> value = std::make_unique<OrtValue>();
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/session/onnxruntime_c_api.cc [hunk 20]
```diff
@@ -958,11 +958,11 @@ static OrtStatus* OrtCreateValueImplSeqHelper(OrtValue** const in, size_t num_va
     vec_ptr->push_back(*data);
   }
   // create MLValue with this vector
-  std::unique_ptr<MLValue> value = std::make_unique<MLValue>();
+  std::unique_ptr<OrtValue> value = std::make_unique<OrtValue>();
```
Comment (RyanUnderhill) [inline onnxruntime/core/session/onnxruntime_c_api.cc]:
```suggestion
  auto value = std::make_unique<OrtValue>();
```

File: onnxruntime/core/framework/execution_frame.h (hunk 5 above)
Comment (snnn) [reply onnxruntime/core/framework/execution_frame.h]: Parameter renamed.  Will rename function later.

**Comment by snnn** [comment]
For all the 'auto' usages, I will do it in a separate, on all existing source files.

**Pull request merged**